daffodil-core/src/main/scala/org/apache/daffodil/core/runtime1/TermRuntime1Mixin.scala

```diff
@@ -332,24 +339,27 @@ trait TermRuntime1Mixin { self: Term =>
           res.getOrElse(Open(Nil))
         }
       }
-      val res = thisItself match {
-        //
-        // check this case separately first to avoid evaluating
-        // followingLexicalSiblingStreamingUnparserElements unless
-        // we have to.
-        //
-        case Closed(pnes1) => thisItself // most common case - a required element.
-        case _ => {
-          val res: PossibleNextElements =
-            (thisItself, followingLexicalSiblingStreamingUnparserElements) match {
-              case (Open(pnes1), Closed(Nil)) =>
-                thisItself // case of Open(...) followed by end of complex element.
-              case (poss1, Closed(pnes2)) => Closed((poss1.pnes ++ pnes2).distinct)
-              case (poss1, poss2) => Open((poss1.pnes ++ poss2.pnes).distinct)
-            }
-          res
+      // System.err.println("thisItself: " + thisItself)
+      // System.err.println("following: " + followingLexicalSiblingStreamingUnparserElements)
+      val res: PossibleNextElements =
+        (thisItself, followingLexicalSiblingStreamingUnparserElements) match {
+          case (poss1, Closed(Nil)) => {
+            thisItself // case of Open(...) followed by end of complex element.
+          }
+          case (Closed(pnes1), Open(List())) => {
+            thisItself // case of Closed(...) followed by empty list, which is default for nothing
+          }
+          case (Closed(pnes1), Closed(pnes2)) => {
+            thisItself // When everything is closed, we only want thisItself and not the entire list of Closed possibilities
+          }
+          case (Open(pnes1), Closed(pnes2)) => {
+            // If there are optional elements before a required element, the whole sequence is considered closed
+            Closed((pnes1 ++ pnes2).distinct)
+          }
+          case (poss1, poss2) => {
```

Comment:
This case is ```case (Open(pnes1), Open(pnes2)) => ```
It would be better to write it that way. Scala *should* detect that the match case is exhaustive.

Reply:
Actually, leaving it like this also absorbs the DoNotUse cases, which matches the behavior we had previously.